src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -842,23 +842,24 @@ RequestContext<object> requestContext
 
             if (parameterSets != null)
             {
-                var sigs = new List<SignatureInformation>();
-                foreach (ParameterSetSignature sig in parameterSets.Signatures)
+                signatures = new SignatureInformation[parameterSets.Signatures.Length];
+                for (int i = 0; i < signatures.Length; i++)
                 {
-                    var parameters = new List<ParameterInformation>();
-                    foreach (ParameterInfo paramInfo in sig.Parameters)
+                    var parameters = new ParameterInformation[parameterSets.Signatures[i].Parameters.Count()];
+                    int j = 0;
+                    foreach (ParameterInfo param in parameterSets.Signatures[i].Parameters)
```

Comment:
Can you use another for loop here instead of foreach since you need the index j anyway?